<launch>
    <!-- map fata -->
    <arg name="map_file0" default="$(find map_switching)/maps/tsukuba/map1.yaml" />
    <arg name="map_file1" default="$(find map_switching)/maps/tsukuba/map2.yaml" />
    <arg name="map_file2" default="$(find map_switching)/maps/tsukuba/map3.yaml" />
    <arg name="map_file3" default="$(find map_switching)/maps/tsukuba/map4.yaml" />
    <arg name="map_file4" default="$(find map_switching)/maps/tsukuba/map5.yaml" />
    <arg name="map_file5" default="$(find map_switching)/maps/tsukuba/map6.yaml" />

<!--    <node pkg="tf" type="static_transform_publisher" name="rs_odom_transform" args="0 0 0 0 0 0 rs_odom rs_odom_frame 100"/>-->
<!--	<node pkg="tf" type="static_transform_publisher" name="stf_hokuyo3d_laser" args="0.225 -0.255 0.0 0.0 0.0 0.0 camera_link 2d_blue 100"/>
	<node pkg="tf" type="static_transform_publisher" name="stf_hokuyo3d_laser2" args="-0.485 0.255 0.0 3.14159 0.0 0.0 camera_link 2d_orange 100"/>-->	
<!--	<node pkg="tf" type="static_transform_publisher" name="rs_base_link_transform" args="-0.2 0 0 0 0 0 rs_link rs_camera_link 100"/>-->
    
    <node pkg="tf" type="static_transform_publisher" name="map_transform" args="0 0 0 0 0 0 map base_link 100"/>

    <!-- map server -->
	<node respawn="true" name="map_server0" pkg="map_server" type="map_server" args="$(arg map_file0)">
        <remap from="/map" to="/map0" />
    </node>
	<node respawn="true" name="map_server1" pkg="map_server" type="map_server" args="$(arg map_file1)">
        <remap from="/map" to="/map1" />
        <remap from="/static_map" to="/static_map1" />
    </node>
	<node respawn="true" name="map_server2" pkg="map_server" type="map_server" args="$(arg map_file2)">
        <remap from="/map" to="/map2" />
        <remap from="/static_map" to="/static_map2" />
    </node>
	<node respawn="true" name="map_server3" pkg="map_server" type="map_server" args="$(arg map_file3)">
        <remap from="/map" to="/map3" />
        <remap from="/static_map" to="/static_map3" />
    </node>
	<node respawn="true" name="map_server4" pkg="map_server" type="map_server" args="$(arg map_file4)">
        <remap from="/map" to="/map4" />
        <remap from="/static_map" to="/static_map4" />
    </node>
    <node respawn="true" name="map_server5" pkg="map_server" type="map_server" args="$(arg map_file5)">
        <remap from="/map" to="/map5" />
        <remap from="/static_map" to="/static_map5" />
    </node>

    <!-- map_switch -->
	<node pkg="map_switching" type="map_selecter" name="map_selecter" output="screen"/>

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find map_switching)/rviz/map_switching.rviz"/>
</launch>